lib/install/rust.js

```diff
@@ -0,0 +1,557 @@
+  }
+}
+
+function sha256stream() {
+  var sha256 = createHash('sha256');
+  var stream = new Transform();
+  stream._transform = function(chunk, encoding, callback) {
+    this.push(chunk);
+    sha256.update(chunk);
+    callback();
+  };
+  stream.on('finish', () => {
+    stream.emit('sha256', sha256.digest('hex'));
+  });
+  return stream;
+}
+
+function sha256file(hash, name) {
+  return `${hash}  ${name}\n`;
+}
+
+function download(url) {
+  var req = request.get(url);
+
+  // When we receive the response
+  req.on('response', (res) => {
+
+    // Parse out the length of the incoming bundle
+    var contentLength = parseInt(res.headers['content-length'], 10);
+
+    // Create a new progress bar
+    var bar = new Progress('     [:bar] :percent :etas remaining', {
+      clear: true,
+      complete: '=',
+      incomplete: ' ',
+      width: 20,
+      total: contentLength
+    });
+
+    // When we get incoming data, update the progress bar
+    res.on('data', (chunk) => {
+      bar.tick(chunk.length);
+    });
+  });
+
+  return req;
+}
+
+function downloadString(url) {
+  return new Promise((resolve, reject) => {
+    request({
+      url,
+      // We want to force Cloudfront to serve us the latest file.
+      headers: {
+        'Accept-Encoding': 'gzip, deflate',
+      },
+    }, (error, response, body) => {
+      if (!error && response.statusCode === 200) {
+        resolve(body);
+      } else {
+        reject(error || response.statusCode);
+      }
+    });
+  });
+}
+
+function tmpdir() {
+  return new Promise((resolve) => {
+    var dir = fsTemp.template('t2-sdk-%s').mkdirSync();
+    resolve({
+      path: dir,
+      cleanup: () => {
+        try {
+          fs.removeSync(dir);
+        } catch (e) {
+          // Swallow errors in removing temporary folder. If the folder was
+          // successfully or unsuccessfully moved, it may not exist at its
+          // location in the temporary directory, but this isn't fatal.
+        }
+      }
+    });
+  });
+}
+
+module.exports.toolchainPath = () => {
+  return new Promise((resolve, reject) => {
+    var sdkPlatformPath = path.join(SDK_PATHS.sdk, getPlatform());
+    var values = fs.readdirSync(sdkPlatformPath);
+
+    for (var i = 0; i < values.length; i++) {
+      if (values[i].match(/^toolchain\-/)) {
+        return resolve(path.join(sdkPlatformPath, values[i]));
+      }
+    }
+    return reject(new Error('No toolchain found.'));
+  });
+};
+
+// Checks if CHECKSUM file in our SDK equals our expected checksum.
+// This will resolve with checking that the SDK exists and matches the checksum.
+module.exports.checkTools = (checksumVerify) => {
+  var dir = path.join(SDK_PATHS.sdk, getPlatform());
+  return new Promise((resolve) => {
+    var checksum = fs.readFileSync(path.join(dir, 'CHECKSUM'), 'utf-8');
+    resolve({
+      exists: true,
+      checked: checksumVerify === checksum,
```

Comment:
If the operation is "verification" and whether or not the tools are verified, then the property should probably be `isVerified`, vs. `checked` (checked doesn't indicate success or failure, ie. I could check something, but the result could've been failure)